pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1968,12 +1978,26 @@ protected void handleGetTopicsOfNamespace(CommandGetTopicsOfNamespace commandGet
                 if (isAuthorized) {
                     getBrokerService().pulsar().getNamespaceService().getListOfTopics(namespaceName, mode)
                         .thenAccept(topics -> {
+                            boolean filterTopics = enableSubscriptionPatternEvaluation && topicsPattern.isPresent()
+                                    && topicsPattern.get().length() <= maxSubscriptionPatternLength;
```

Comment:
Should we provide some way to expose the situation if `topicsPattern.get().length() <= maxSubscriptionPatternLength`? Or else it would very hard for user to know why topic pattern not working .

Reply:
> From a user perspective there would be no visible difference.

The user here is more like cluster maintainer who have access to broker config to enable this feature.



> The point is to avoid "unbound" regexes on server side.

Yes, this limitation is reasonable. But I think we need a warn log here for cluster maintainer, as they may not know exactly how long the topic pattern string is used.